packages/bugsnag_flutter/android/src/main/java/com/bugsnag/flutter/BugsnagFlutter.java

```diff
@@ -406,6 +409,23 @@ JSONObject createEvent(@Nullable JSONObject args) throws JSONException {
             event.addMetadata("flutter", (Map<String, Object>) flutterMetadata);
         }
 
+        JSONObject correlation = args.optJSONObject("correlation");
+        if (correlation != null) {
+
+            String traceId = getString(correlation, "traceId");
+            String spanId = getString(correlation, "spanId");
+            if (traceId != null &&
+                    traceId.length() == HEX_LONG_LENGTH * 2 &&
+                    spanId != null &&
+                    spanId.length() == HEX_LONG_LENGTH
+            ) {
+                long traceIdMostSignificantBits = hexToLong(traceId.substring(0, HEX_LONG_LENGTH));
+                long traceIdLeastSignificantBits = hexToLong(traceId.substring(HEX_LONG_LENGTH));
+                long spanIdAsLong = hexToLong(spanId);
```

Comment:
we should do some error-catching (`NumberFormatException` at minimum) here, if (somehow) these aren't exactly what we expect we would crash on the hot-path... if the strings aren't hex encoded we can safely omit the entire `correlation` element